lib/commands/pkg.js

```diff
@@ -61,12 +57,12 @@ class Pkg extends BaseCommand {
     }
   }
 
-  async pkgWorkspaces (args, filters) {
+  async execWorkspaces (args, filters) {
     await this.setWorkspaces(filters)
     const result = {}
     for (const [workspaceName, workspacePath] of this.workspaces.entries()) {
       this.prefix = workspacePath
-      result[workspaceName] = await this.pkg(args)
+      result[workspaceName] = await this.exec(args, { prefix: workspacePath })
```

Comment:
It's a little more subtle than that.  This passes it in to exec, which defaults its second parameter to `{}`, so that when npm calls exec (and doesn't give that second param) `prefix` isn't set and is set from this.npm.localPrefix

```
  async exec (args, { prefix } = {}) {
    if (!prefix)
      this.prefix = this.npm.localPrefix
    else
      this.prefix = prefix
```

It's a little goofy but this was essentially a 1:1 logical port of how it worked before. Probably a better way to do this but it felt outside the scope of this refactor.